### PR TITLE
[SMALLFIX] Use deploy mode to configure zookeeper

### PR DESCRIPTION
**Pull request by aaudiber**

### CODE REVIEW

File: minicluster/src/main/java/alluxio/multi/process/MultiProcessCluster.java [hunk 1]
```diff
@@ -121,15 +123,21 @@ public synchronized void start() throws Exception {
...
-      mProperties.put(PropertyKey.MASTER_RPC_PORT, Integer.toString(masterAddress.getRpcPort()));
-      mProperties.put(PropertyKey.MASTER_WEB_PORT, Integer.toString(masterAddress.getWebPort()));
+    switch (mDeployMode) {
+      case ZOOKEEPER_HA:
+        mCuratorServer = mCloser
+            .register(new TestingServer(-1, AlluxioTestDirectory.createTemporaryDirectory("zk")));
+        mProperties.put(PropertyKey.ZOOKEEPER_ENABLED, "true");
+        mProperties.put(PropertyKey.ZOOKEEPER_ADDRESS, mCuratorServer.getConnectString());
+        break;
+      case NON_HA:
```
Comment (calvinjia) [inline minicluster/src/main/java/alluxio/multi/process/MultiProcessCluster.java]: Could you put these in alphabetical order?

File: minicluster/src/main/java/alluxio/multi/process/MultiProcessCluster.java [hunk 2]
```diff
@@ -449,6 +463,10 @@ private Builder() {} // Should only be instantiated by newBuilder().
      * @return the builder
      */
     public Builder addProperty(PropertyKey key, String value) {
+      if (key.equals(PropertyKey.ZOOKEEPER_ENABLED)) {
+        throw new IllegalArgumentException(
+            "Enable Zookeeper via #setDeployMode instead of #addProperty");
+      }
```
Comment (calvinjia) [inline minicluster/src/main/java/alluxio/multi/process/MultiProcessCluster.java]: Precondition instead?

File: minicluster/src/main/java/alluxio/multi/process/MultiProcessCluster.java (hunk 1 above)
Comment (aaudiber) [reply minicluster/src/main/java/alluxio/multi/process/MultiProcessCluster.java]: done

File: minicluster/src/main/java/alluxio/multi/process/MultiProcessCluster.java (hunk 2 above)
Comment (aaudiber) [reply minicluster/src/main/java/alluxio/multi/process/MultiProcessCluster.java]: done

**Comment by calvinjia** [comment]
@aaudiber lgtm, could you resolve the conflicts?

**Pull request merged**